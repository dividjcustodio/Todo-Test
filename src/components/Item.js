import React from "react";
import "../App.css";
import { ListItem, ListItemText  } from '@material-ui/core';
import { Link } from "react-router-dom";
import useStyles from "../assets/styles";

function ItemComponent(item) {
  const classes = useStyles();
  return (
    <ListItem data-testid="item">
      <Link
        key={`ind-${item.id}`}
        to={`/detail/${item.id}`}
      >
        {item.title}
      </Link>
      {item.completed ? 
        <ListItemText className={`${classes.greenFont} ${classes.flexEnd}`} primary="completed" />:
        <ListItemText className={`${classes.redFont} ${classes.flexEnd}`} primary="no completed" />
      }
    </ListItem>
  );
}

export default ItemComponent;
