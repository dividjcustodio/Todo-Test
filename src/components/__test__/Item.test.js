import React from "react";
import Item from "./../Item";
import "@testing-library/jest-dom";
import Adapter from "enzyme-adapter-react-16";
import { cleanup } from "@testing-library/react";
import { createShallow, createRender } from '@material-ui/core/test-utils';
import {configure} from "enzyme";
import Enzyme from "enzyme/build";
import { MemoryRouter } from 'react-router'
configure({ adapter: new Adapter() });
Enzyme.configure({ adapter: new Adapter() });


describe('>>>Pages --- Snapshot', () => {
  let shallow = createShallow();
  let render = createRender();

  afterEach(cleanup);
  it("normal rendering", () => {
    render(<MemoryRouter><Item></Item></MemoryRouter>);
  });

  it("renders Items", () => {
    const dataItem = {
      userId: 1,
      id: 1,
      title: 1,
      completed: false
    };
    render(<MemoryRouter><Item {...dataItem} /></MemoryRouter>);
  });
  const ItemContainer = shallow(<MemoryRouter><Item userId={1} id={1} title={"some"} completed={false} /></MemoryRouter>);

  it("+++capturing Snapshot of Item", () => {
    expect(ItemContainer.html()).toMatchSnapshot();
  });


});
