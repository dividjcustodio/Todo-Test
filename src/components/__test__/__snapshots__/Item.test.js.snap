// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Pages --- Snapshot +++capturing Snapshot of Item 1`] = `"<li class=\\"MuiListItem-root MuiListItem-gutters\\" data-testid=\\"item\\"><a href=\\"/detail/1\\">some</a><div class=\\"MuiListItemText-root makeStyles-redFont-8 makeStyles-flexEnd-3\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1\\">no completed</span></div></li>"`;
